### DISCUSSION TITLE
Memory Leak

### DISCUSSION THREAD

**Discussion by Thomas-Lerman** [pelicanmapping/osgearth#2338] [Support Q&A]
On an older version of osgEarth, I have a memory leak that I have not been able to track down yet. I wanted to see if anyone has ran into this in the past. I have a drawing object that gets frequently updated (maybe 1 Hz). I can watch the memory utilization go up and up. I found that if I setNodeMask() to where the drawing object is not display, the memory utilization stops growing. I am hoping someone has ran into this and I can fix it. @emminizer or anyone else, have you by chance seen this? If not, suggestions for tracking this down? I have watched the ref_ptr and it seems to get released, etc.

**Comment by emminizer** [comment]
I have not encountered this, at least it doesn't ring a bell as described.

>  I have a drawing object that gets frequently updated (maybe 1 Hz).

Could you describe better what you mean here, specifically by "drawing object"? Maybe that could help ring a bell. We use osgEarth Geometry, LocalGeometryNode, etc. for rendering shape data and do regularly memcheck those, not sure if that's precisely what you meant.

**Comment by Thomas-Lerman** [comment]
The drawing object may be one or more of the following:
1. osgEarth::Features:Feature containing osgEarth::Symbology::MultiGeometry containing one or more of the following (do not mix them): osgEarth::Symbology::Polygon or osgEarth::Symbology::LineString
2. osgEarth::Annotation::EllipseNode.

**Reply by emminizer** [reply]
Thanks. We do use EllipseNode a lot, and we have several MultiGeometry. Unfortunately this just doesn't ring a bell with me, sorry I don't specifically remember any leak fixes related to these and a quick review of my old emails doesn't show anything very relevant either.
